### ISSUE TITLE
FI translation updated, please move files to correct places ;)

### ISSUE THREAD

**Issue by LaruX75** [knicos/genai-lm#3]
I fixed FI localisation little bit:
* home.json (new one)
* translation.json (updated translation with new variables compared to earlier FI localisation)

----
home.json 
{
    "title": "Generation AI",
    "llmTitle": "Pieni kielikone",
    "llmDescription": "Luo pieniä kielimalleja selaimessasi ja opi kielimallinnuksen prosessista, mekanismeista ja haasteista. Tämä sovellus käyttää samaa taustalla olevaa transformer-tekniikkaa kuin suuret kielimallit, mutta paljon pienemmässä mittakaavassa, joka toimii omalla koneellasi.",
    "activity": "Valitse aktiviteetti aloittaaksesi:",
    "howItWorksTitle": "Miten se toimii...",
    "howItWorksContent": "Tämä sovellus toimii kokonaan selaimessasi käyttäen JavaScriptiä ja WebGL:ää tai WebGPU:ta laitteistokiihdytykseen. Mitään tietoja ei lähetetä palvelimelle; kaikki toimii paikallisesti omalla koneellasi. Sovellus perustuu Andrej Karpathyn NanoGPT:hen, joka on minimalistinen toteutus GPT-arkkitehtuurista (Generative Pre-trained Transformer). Toteutus on optimoitu ja mukautettu selainympäristöön, mutta joidenkin laitteiden suorituskyky voi silti kärsiä käytetyn mallin ja aineiston koosta riippuen.",
    "whoMadeItTitle": "Kuka tämän on tehnyt...",
    "whoMadeItContent": "Tämän sovelluksen on luonut Nicolas Pope osana Generation AI -tutkimushanketta Suomessa. Hanke on saanut julkista rahoitusta tekoälylukutaidon edistämiseksi suomalaisissa kouluissa.",
    "activities": {
        "pretrain": {
            "title": "Kielimallin esikouluttaminen tyhjästä",
            "description": "Kouluta kielimalli tyhjästä omalla aineistollasi. Opi koulutusprosessista ja mekanismeista, jotka liittyvät kielen tuottamiseen."
        },
        "classifier": {
            "title": "Kielimallin hienosäätö luokittelijaksi",
            "description": "Ota esikoulutettu kielimalli ja mukauta se tekstiluokittelijaksi, jonka voit jakaa sovelluksena. Opi, miten dataa merkitään ja jäsennellään ohjattua oppimista varten."
        },
        "chatbot": {
            "title": "Kielimallin hienosäätö chatbotiksi",
            "description": "Rakenna esikoulutetusta kielimallista chatbot-sovellus. Opi strukturoimaan keskusteludataa vuorovaikutteisia dialogeja varten."
        }
    }
}
Translation json
{
    "app": {
        "little": "Pieni",
        "languageMachine": "Kielikone",
        "save": "Tallenna",
        "load": "Avaa",
        "saveTitle": "Tallenna kielimalli",
        "saveMessage": "Anna kielimallille nimi:",
        "saveName": "Kielimallin nimi",
        "saveSave": "Tallenna",
        "saveCancel": "Peruuta",
        "settings": {
            "title": "Asetukset",
            "general": "Yleiset",
            "training": "Opettaminen",
            "generator": "Generointi",
            "showModelSelect": "Näytä kielimallien valinta",
            "showEvaluation": "Näytä arviointi-ikkuna",
            "showTrainer": "Näytä opetusikkuna",
            "showDataEntry": "Näytä dataikkuna",
            "showSettings": "Näytä asetukset",
            "showPrompt": "Kehote (prompt)",
            "showProbabilities": "Näytä todennäköisyydet",
            "showAttention": "Visualisoi huomiomekanismi",
            "maxLength": "Generoidun tekstin enimmäispituus",
            "temperature": "Sattumanvaraisuus (lämpötila)",
            "batchSize": "Tekstinäyte-erän koko",
            "learningRate": "Oppimisnopeus",
            "showOutputText": "Näytä mallin kehittyminen",
            "checkpointing": "Poista gradienttitarkistuspisteet käytöstä",
            "showXAI": "Näytä selittävä tekoäly (XAI)",
            "topP": "Top P -otanta",
            "evaluationMetrics": "Arviointimittarit",
            "qualityMetric": "Laatu",
            "perplexityMetric": "Perpleksisuus",
            "lossMetric": "Häviö (loss)",
            "advancedMetrics": "Edistyneet mittarit",
            "performProbe": "Tarkista laitteisto käynnistyksen yhteydessä",
            "developerMode": "Kehittäjätila",
            "mixedPrecision": "Käytä sekatarkkuutta (Mixed Precision)",
            "lowPowerMode": "Virransäästötila",
            "disableSubgroups": "Poista alaryhmittely (subgrouping) käytöstä"
        },
        "close": "Sulje",
        "ok": "OK",
        "annotations": {
            "model": "Malli",
            "data": "Aineisto"
        },
        "billion": "B",
        "million": "M",
        "thousand": "k"
    },
    "deviceProbe": {
        "probing": "Tarkastellaan laitteistosi suorituskykyä"
    },
    "data": {
        "title": "Opetusaineisto",
        "textDialogTitle": "Tekstisisältö",
        "addTooltip": "Lisää uutena opetusaineistona",
        "dataHint": "Kirjoita tekstiä tai lataa erilaisia tiedostotyyppejä (txt, csv, pdf, docx) opetusaineistoksi. Mitä enemmän tekstiä, sitä parempi.",
        "modelHint": "Valitse ensin malli.",
        "add": "Kirjoita",
        "upload": "Lataa",
        "untitled": "Teksti",
        "addText": "Lisää",
        "save": "Tallenna",
        "cancel": "Peruuta",
        "close": "Sulje",
        "textPlaceholder": "Kirjoita tai liitä oma tekstisi tähän...",
        "dropHint": "Pudota tiedostot tähän...",
        "ok": "OK",
        "samples": "Tekstinäytettä",
        "search": "Hae",
        "downloading": "Ladataan {{count}} tiedosto(a)",
        "errors": {
            "fileLoadError": "Voit pudottaa vain tekstitiedostoja (txt, csv, pdf, docx).",
            "downloadLoadError": "Ladatun tiedoston avaamisessa tapahtui tuntematon virhe."
        }
    },
    "model": {
        "parameters": "Parametrit",
        "memory": "Muisti",
        "title": "Kielimalli",
        "create": "Luo uusi",
        "upload": "Lataa",
        "tools": "Työkalut",
        "share": "Jaa",
        "download": "Lataa",
        "search": "Hae",
        "charTokeniser": "{{size}} merkkiä",
        "bpeTokeniser": "{{size}} sanaketta",
        "layers": "{{count}} kerrosta",
        "contextSize": "Konteksti-ikkunan koko: {{size}}",
        "duration": "{{duration}} opetusta",
        "durationHours": "{{count}} tuntia",
        "durationMinutes": "{{count}} minuuttia",
        "durationSeconds": "{{count}} sekuntia",
        "samples": "{{samples}} tekstinäytettä",
        "embeddingSize": "Sisäinen koko {{size}}",
        "modelHint": "Valitse tai lataa kielimalli.",
        "languageModel": "Kielimalli",
        "visualisation": {
            "next": "Seuraava"
        },
        "preTraining": "Esikoulutus",
        "defaultName": "Oma mallisi"
    },
    "training": {
        "title": "Opettaminen",
        "samples": "Näytettä",
        "start": "Opeta malli",
        "stop": "Pysäytä",
        "hours": "t",
        "minutes": "m",
        "seconds": "s",
        "remaining": "Jäljellä",
        "settings": "Asetukset",
        "loss": "Häviö (loss)",
        "valLoss": "Validointihäviö (validation loss)",
        "steps": "Askelta",
        "noData": "Opetusaineistoa ei ole saatavilla",
        "monitor": "Seuranta",
        "visualize": "Visualisointi",
        "autoOutput": "Näytä mallin kehittyminen",
        "errors": {
            "preparation": "Tarvitaan lisää opetusaineistoa, vähintään muutama tuhat tekstinäytettä.",
            "noModel": "Valitse tai lataa malli ennen opettamista.",
            "notReady": "Malli tai aineisto ei ole valmis opettamiseen.",
            "noData": "Lisää ensin opetusaineistoa.",
            "trainingFailed": "Mallin opettaminen epäonnistui. Päivitä sivu ja kokeile pienempää mallia tai erän kokoa (batch size)."
        }
    },
    "generator": {
        "title": "Tekstin luominen",
        "generate": "Luo tekstiä",
        "promptPlaceholder": "Valinnainen kehoteteksti...",
        "settingsTooltip": "Generoinnin asetukset",
        "pause": "Pysäytä",
        "editPlaceholder": "Kirjoita halutessasi kehoteteksti tähän...",
        "reset": "Tyhjennä",
        "copy": "Kopioi leikepöydälle",
        "autoGenerateAriaLabel": "Vaihda automaattinen generointi päälle/pois",
        "autoGenerate": "Automaattinen generointi",
        "step": "Seuraava",
        "errors": {
            "modelNotReady": "Malli ei ole valmis generointiin, yritä hetken kuluttua uudelleen.",
            "generationError": "Tekstin generoinnissa tapahtui tuntematon virhe."
        }
    },
    "evaluation": {
        "title": "Laatu",
        "quality": "Laatu",
        "perplexity": "Perplexisuus",
        "loss": "Häviö",
        "samplesPerSecond": "Näytteitä/s",
        "memory": "Muisti",
        "gradientNorm": "Gradientin normi"
    },
    "modelStatus": {
        "warmup": "Mallia valmistellaan...",
        "ready": "Malli on valmis käyttöön",
        "training": "Mallia opetetaan...",
        "loading": "Mallia ladataan...",
        "busy": "Malli käsittelee dataa...",
        "error": "Mallissa tapahtui virhe",
        "none": "Mallia ei ole ladattu",
        "awaitingTokens": "Odotetaan sanakkeita (token)..."
    },
    "validation": {
        "invalid_number": "Virheellinen numero",
        "out_of_range": "Arvon on oltava välillä {{min}} ja {{max}}",
        "required": "Tämä kenttä on pakollinen",
        "positive_integer": "Arvon on oltava positiivinen kokonaisluku",
        "unknown_error": "Tuntematon virhe tapahtui",
        "nEmbed_divisible_nHead": "Upotuksen koon on oltava jaollinen huomiokärkien määrällä",
        "headDim_even": "Kokeile toista kärkien määrää"
    },
    "xai": {
        "title": "Selitys"
    },
    "tools": {
        "Settings": "Asetukset",
        "Visualizations": "Visualisoinnit",
        "generatorSettings": "Generoinnin asetukset",
        "trainingSettings": "Opettamisen asetukset",
        "checks": "Operaattoritestit",
        "advanced": "Edistyneet työkalut",
        "debugModel": "Debuggaa malli",
        "nextSample": "Seuraava",
        "trainingProcess": "Opetusprosessi",
        "loss": "Häviö (loss)",
        "model": "Malli",
        "modelMissingHint": "Lataa malli ja opetusaineisto nähdäksesi opetusprosessin.",
        "play": "Toista",
        "pause": "Tauko (pause)",
        "step": "Askel",
        "training": {
            "title": "Opetuksen loki",
            "validation": "Validointi",
            "history": "Historia",
            "performance": "Suorituskyky"
        },
        "modelPredicting": "Ennustetaan...",
        "modelUpdating": "Päivitetään parametreja...",
        "modelComplete": "Opetusaskel valmis",
        "modelIdle": "Odotetaan...",
        "nlayers": "{{N}} kerrosta",
        "gradients": "Gradientit"
    }
}

**Comment by knicos** [comment]
There was still a lot missing so I used AI. You might want to check it sometime.

**Comment by LaruX75** [comment]
Yep. I will do!
